src/auth/strategy/apikey.strategy.ts

```diff
@@ -0,0 +1,20 @@
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { PassportStrategy } from '@nestjs/passport';
+import { HeaderAPIKeyStrategy } from 'passport-headerapikey';
+
+import { AuthService } from '../auth.service';
+
+@Injectable()
+export class ApiKeyStrategy extends PassportStrategy(HeaderAPIKeyStrategy, 'x-api-key') {
+  constructor(private authService: AuthService, private configService: ConfigService) {
+    const headerKeyApiKey: string = configService.get<string>('app.apiKeyStrategy');
+    super({ header: headerKeyApiKey, prefix: '' }, true, (apiKey: string, done: (arg0: UnauthorizedException, arg1: boolean) => void) => {
+      const hasValidKey: boolean = this.authService.validateApiKey(apiKey);
+      if (hasValidKey) {
+        done(null, true);
+      }
+      done(new UnauthorizedException(), null);
+    });
```

Comment:
It would be nice to unit test this behaviour, although I imagine it would be slightly fiddly because it is in the constructor of a 3rd party class.

I think in this case we might be able to:

1. mock the HeaderAPIKeyStrategy constructor, and mock the auth service and config service
2. call the ApiKeyStrategy constructor with the mocked auth service and config service
3. test that the first 2 inputs to the HeaderAPIKeyStrategy constructor are as expected
4. get the 3rd input to the `HeaderAPIKeyStrategy` constructor (the `verify` function) and call it with an `apiKey` and mocked `done` function - then we can test that it validates the api key correctly

Reply:
I'll address this once we refactor the whole test work